Test/dafny0/AutoContracts.dfy.expect

```diff
@@ -31,7 +31,7 @@ module _System {
   type {:axiom} array(==)<arg> = c: array?<arg> | c != null /*special witness*/
   */
 
-  class {:compile false} /*_#Func1*/ -T0 ~> +R {
+  type {:compile false} /*_#Func1*/ -T0 ~> +R {
```

Comment:
Seems like this PR is more than just a refactoring

Reply:
This change is because `ArrowTypeDecl` used to inherit from `ClassDecl` and now inherits from `ValuetypeDecl`. Really, arrow types were never classes in the first place.